Multi monitor fix

Cecer opened the pull request:
**Edit:** See [Quantumheart's comment](https://github.com/Ttanasart-pt/xskills-gilded/pull/1#issuecomment-3429861363) below for a cleaner fix.

---

This commit fixes the issue where the window position is incorrect when opened on a system with multiple monitors where the game was not in the top-left most monitor.

The issue appeared to be that `window.absOffsetX/Y` both always returned 0.

I could not find a way to get the window location within Vintage Story codebase directly but this information can be accessed with OpenTK. As such I had to add a reference to OpenTK.

I suspect there might also be issues with `LevelPopup, EffectBox` but I have not looked into that.

I also replaced `(int)window.OuterWidth` with `window.OuterWidthInt` (and same with height) as I noticed that was an option.

FYI: This has only been tested on Linux as I do not have any other operation systems set up for testing.

Cecer committed:
Fix window position with multiple monitors

Cecer committed:
Use OuterWidth/HeightInt instead of casting

Cecer commented:
I can confirm this does not fix LevelPopup or EffectBox. I have not had time to work on fixes for those though.

Quantumheart commented:
@Cecer I fixed this my multi monitor issues with the the `ImGui.GetMainViewport();` method.

Level Popup change
https://github.com/Quantumheart/xskills-gilded/commit/d1766894c833cae0c491b4bd92fd5ff81662a747

DeadSigma commented:
> @Cecer I fixed this my multi monitor issues with the the `ImGui.GetMainViewport();` method.
> 
> Level Popup change [Quantumheart@d176689](https://github.com/Quantumheart/xskills-gilded/commit/d1766894c833cae0c491b4bd92fd5ff81662a747)

Huge thanks for this solution! Using ImGui.GetMainViewport() worked perfectly and fixed the multi-monitor issue as well as the LevelPopup problem. I’ve applied this approach to the main code.
I’m currently maintaining the mod on the mod site while @Ttanasart-pt  is away, but I’ll always be happy to see the creator return!

Cecer commented:
> @Cecer I fixed this my multi monitor issues with the the `ImGui.GetMainViewport();` method.
> 
> Level Popup change [Quantumheart@d176689](https://github.com/Quantumheart/xskills-gilded/commit/d1766894c833cae0c491b4bd92fd5ff81662a747)

That looks much nicer. Thanks!



> > @Cecer I fixed this my multi monitor issues with the the `ImGui.GetMainViewport();` method.
> > Level Popup change [Quantumheart@d176689](https://github.com/Quantumheart/xskills-gilded/commit/d1766894c833cae0c491b4bd92fd5ff81662a747)
> 
> Huge thanks for this solution! Using ImGui.GetMainViewport() worked perfectly and fixed the multi-monitor issue as well as the LevelPopup problem. I’ve applied this approach to the main code. I’m currently maintaining the mod on the mod site while @Ttanasart-pt is away, but I’ll always be happy to see the creator return!

Wonderful! Thank you too :+1:

Ttanasart-pt commented:
I haven't played VS for a while now, but this seems to be a correct solution. I wish there were an easier way to get something as simple as window position.
